great_expectations/validator/validator.py

```diff
@@ -1511,6 +1492,37 @@ def validate(  # noqa: C901, PLR0912, PLR0913, PLR0915
 
         return result
 
+    def process_expectations_for_validation(
+        self,
+        expectation_configurations: list[ExpectationConfiguration],
+        evaluation_parameters: Optional[dict[str, Any]] = None,
+    ) -> list[ExpectationConfiguration]:
+        """Substitute evaluation parameters into the provided expectations and sort by column."""
+        columns: dict[Any, list[ExpectationConfiguration]] = {}
+
+        for expectation in expectation_configurations:
+            expectation.process_evaluation_parameters(
+                evaluation_parameters=evaluation_parameters,
+                interactive_evaluation=self.interactive_evaluation,
+                data_context=self._data_context,
+            )
```

Comment:
Could we do all this work on the actual expectation instead of the config? Perhaps we need to shift the logic from the config to the actual expectation?

Reply:
Also what is the context dependency?

Reply:
Most of the heavy lifting here is still on the expectation, since we are just calling `expectation.process_evaluation_parameters`. The sorting below likely needs to live outside of the expectation (unless it were a class method). But overall my goal here was just move this logic from the above `validate` method so that we can reuse it without doing a more heavy refactor.

Right now context is only used (more deep in the call stack in `parse_evaluation_parameter`) to reach into the metrics store. That can come out, but will require some bigger changes that I think we should tackle in our upcoming work on evaluation params.

I'd like to argue that this change in this PR should stay as is because a) this is in the old Validator that we have a shim over top of, and it's not degrading the quality of any of our abstractions (since it's just cut/paste within the same class). LMK if you disagree though!